doc: Updating README.md, submitter scripts in example2

cmoussa1 opened the pull request:
This PR introduces minor changes to the following files:

1. **README.md**
- A table listing of the equivalent commands for setting FLUX_SCHED_OPTIONS in both tcsh and bash/zsh shells.
- Numbers for each step in the sub-examples instead of bullet points.
- Markdown code format for the command line steps in each example instead of bold font.

2. **submitter.py** and **submitter2.py**
- When running the original scripts as is, I would get a "failed" job state when running `flux wreck ls`; see example below:

```
bash-4.2$ flux wreck ls
    ID NTASKS STATE                    START      RUNTIME    RANKS COMMAND
     1      4 failed     2019-10-07T10:14:49       0.174s    [0-1] compute.py
     2      1 failed     2019-10-07T10:14:49       0.137s        2 io-forwarding
```

Adding `./` in front of the scripts to be run in the `cmdline` key-value pair in both dictionaries fixed the problem, and running `flux wreck ls` showed successful "running" and "exited" job states:

```
bash-4.2$ flux wreck ls
    ID NTASKS STATE                    START      RUNTIME    RANKS COMMAND
     2      1 running    2019-10-07T10:15:44       2.291s        2 io-forwarding
     1      4 running    2019-10-07T10:15:44       2.325s    [0-1] compute.py
```
```
bash-4.2$ flux wreck ls
    ID NTASKS STATE                    START      RUNTIME    RANKS COMMAND
     1      4 exited     2019-10-07T10:15:44       2.031m    [0-1] compute.py
     2      1 exited     2019-10-07T10:15:44       2.027m        2 io-forwarding
```

3. **submitter.lua** and **submitter2.lua**

Same changes as described in 2.

cmoussa1 commented:
Oops, did I mess up somewhere? Looks like this PR also has the commits from PR #14 from last Friday, even though I submitted this one from a different branch (example2-update).

dongahn commented:
@cmoussa1: why don't I swing by your office.

cmoussa1 commented:
Okay sounds good.

cmoussa1 commented:
I talked with Dong; it's just that I needed to clarify that on this PR, PR #14, needs to be merged first before merging this one.

dongahn commented:
@cmoussa1: please rebase to the upstream/master and force a push. If your remote name for the upstream is "upstream", `git fetch upstream; git rebase upstream/master" should do. There should be no conflict.

cmoussa1 commented:
Okay, I think I rebased and force pushed correctly.

FWIW, I rebased from `flux-core-0.11.flux-sched-0.7`, not `master`, since that's the branch where we have been pushing my changes. So the following commands is the sequence I followed:

```
> git fetch upstream
> git rebase upstream/flux-core-0.11.flux-sched-0.7
> git push -f origin example2-update
Counting objects: 14, done.
Delta compression using up to 8 threads.
Compressing objects: 100% (14/14), done.
Writing objects: 100% (14/14), 2.20 KiB | 2.20 MiB/s, done.
Total 14 (delta 11), reused 0 (delta 0)
remote: Resolving deltas: 100% (11/11), completed with 6 local objects.
To https://github.com/cmoussa1/flux-workflow-examples.git
 + d5e4124...44f5e64 example2-update -> example2-update (forced update)
```

dongahn commented:
> FWIW, I rebased from flux-core-0.11.flux-sched-0.7, not master, since that's the branch where we have been pushing my changes. So the following commands is the sequence I followed:

Ah, yes, that's what you should do. Thanks!

dongahn commented:
Looks great. I'm merging. 

As a future reference, if you see an issue which required a fix, one could open up an issue to track the history better. For the minor problems like adding `./`, it probably isn't worthy it. But if your bug fix is significant, that would be a good practice.

cmoussa1 commented:
Yes, good point; I will keep this in mind for the future!
